src/badges.js

```diff
@@ -220,14 +222,51 @@ function findNatureDOIs() {
           doi: decodeURIComponent(doi[1])
         })
       }
+      break
     }
   }
   return els
 }
 
 
 /**
- * findGoogleScholarDOIs looks in reference tags that link to doi.org.
+ * findSpringerDOIs looks in reference tags that link to doi.org.
+ * @returns {Array<{ citeEl: HTMLElement, doi: string}>} - Return
+ */
+function findSpringerDOIs() {
+  const els = []
+  let cites = document.body.querySelectorAll('.c-article-references__links')
+  for (let cite of cites) {
+    const anchors = cite.querySelectorAll('a')
+    for (let anchor of anchors) {
+      const doi = anchor.href.match(/doi\.org\/(.+)/)
+      if (doi && doi.length > 1) {
+        els.push({
+          citeEl: cite,
+          doi: decodeURIComponent(doi[1])
+        })
+      }
+    }
+  }
+  cites = document.body.querySelectorAll('.c-reading-companion__reference-item')
+  for (let cite of cites) {
+    const anchors = cite.querySelectorAll('a')
+    for (let anchor of anchors) {
+      const doi = anchor.href.match(/doi\.org\/(.+)/)
+      if (doi && doi.length > 1) {
+        els.push({
+          citeEl: cite,
+          doi: decodeURIComponent(doi[1])
+        })
+      }
+      break
+    }
+  }
+  return els
+}
```

Comment:
Could this be refactored to share code so it would be:

```
const cites_1 = querySelectorAll(...)
const cites_2 = querySelectorAll(...)
const els = [...getElsFromAnchors(cites_1), ...getElsFromAchors(cites_2)]
return els
```
Or similar